plugins/SegmentEditor/SegmentEditor.php

```diff
@@ -71,6 +85,92 @@ public function getKnownSegmentsToArchiveForSite(&$segments, $idSite)
+        // be confused.
+        $model = new Model();
+        $storedSegment = $model->getSegmentByDefinition($segment->getString()) ?: null;
+
+        throw new UnprocessedSegmentException($segment, $isSegmentToPreprocess, $storedSegment);
+    }
+
+    public function onLoadingError(\Exception $ex, View $dataTableView)
+    {
+        if (!($ex instanceof UnprocessedSegmentException)) {
+            Notification\Manager::cancel(self::NO_DATA_UNPROCESSED_SEGMENT_ID);
+            return;
+        }
+
+        $segment = $ex->getSegment();
+        $segmentToPreprocess = $ex->getStoredSegment();
+
+        $segmentDisplayName = !empty($segmentToPreprocess['name']) ? $segmentToPreprocess['name'] : $segment->getString();
+
+        $view = new View('@SegmentEditor/_unprocessedSegmentMessage.twig');
+        $view->isSegmentToPreprocess = $ex->isSegmentToPreprocess();
+        $view->segmentName = $segmentDisplayName;
+        $view->visitorLogLink = '#' . Url::getCurrentQueryStringWithParametersModified([
+            'category' => 'General_Visitors',
+            'subcategory' => 'Live_VisitorLog',
+        ]);
+
+        $notification = new Notification($view->render());
+        $notification->priority = Notification::PRIORITY_HIGH;
+        $notification->context = Notification::CONTEXT_INFO;
+        $notification->flags = Notification::FLAG_NO_CLEAR;
+        $notification->type = Notification::TYPE_TRANSIENT;
+        $notification->raw = true;
+
+        unset($dataTableView->error); // don't display the API error
+        $dataTableView->notifications[self::NO_DATA_UNPROCESSED_SEGMENT_ID] = $notification;
+    }
+
+    private function getSegmentIfIsUnprocessed()
+    {
+        // TODO: if in cron archiving, return null;
+        // get idSites
+        $idSite = Common::getRequestVar('idSite', false);
+        if (empty($idSite)
+            || !is_numeric($idSite)
+        ) {
+            return null;
+        }
+
+        // get segment
+        $segment = Common::getRequestVar('segment', false);
+        if (empty($segment)) {
+            return null;
+        }
+        $segment = new Segment($segment, [$idSite]);
+
+        // get period
+        $date = Common::getRequestVar('date', false);
+        $period = Common::getRequestVar('period', false);
+        $period = Period\Factory::build($period, $date);
+
+        // check if archiving is enabled. if so, the segment should have been processed.
+        $isArchivingDisabled = Rules::isArchivingDisabledFor([$idSite], $segment, $period);
+        if (!$isArchivingDisabled) {
+            return null;
+        }
+
+        // check if requested segment is segment to preprocess
+        $segmentsToPreprocess = SettingsPiwik::getKnownSegmentsToArchive();
+        $segmentsToPreprocess = array_merge($segmentsToPreprocess, SettingsPiwik::getKnownSegmentsToArchiveForSite($idSite));
+        $isSegmentToPreprocess = in_array($segment, $segmentsToPreprocess);
```

Comment:
I wonder if we also need to try `in_array(urlencode($segment), $segmentsToPreprocess)` and/or `in_array(urldecode($segment), $segmentsToPreprocess)` as I know we accept it somehow encoded and not encoded or so.

Reply:
It might be similar to `Rules::isSegmentPreProcessed()`

Reply:
I think `Segment` will urldecode, so if it's in the config/DB encoded I suppose somehow it could work. I can add the check for that.